misc/keygen/src/main.rs

```diff
@@ -0,0 +1,126 @@
+use std::error::Error;
+use std::path::PathBuf;
+use std::str::{self, FromStr};
+use std::sync::mpsc;
+use std::thread;
+
+mod config;
+
+use libp2p::identity::{self, ed25519};
+use libp2p::PeerId;
+use structopt::StructOpt;
+use zeroize::Zeroizing;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "libp2p key material generator")]
+struct Args {
+    /// JSON formatted output
+    #[structopt(long, global = true)]
+    json: bool,
+
+    #[structopt(subcommand)]
+    cmd: Command,
+}
+
+#[derive(Debug, StructOpt)]
+enum Command {
+    /// Generate from...
```

Comment:
```suggestion
    /// Read from config file
```


That would make for a nicer `--help` output. Currently:

```
SUBCOMMANDS:
    from    Generate from...
    help    Prints this message or the help of the given subcommand(s)
    rand    Generate random
```